vm/src/vm.rs

```diff
@@ -218,7 +224,7 @@ impl VirtualMachine {
         &self.ctx
     }
 
-    pub fn get_builtin_scope(&mut self) -> ScopeRef {
+    pub fn get_builtin_scope(&self) -> ScopeRef {
```

Comment:
Less mut is better!

Reply:
I suspect there's lots of mut we could get rid of on the VirtualMachine.